src/read.c

```diff
@@ -1953,26 +1953,34 @@ static avifBool avifParseImageGridBox(avifImageGrid * grid,
 
 #if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
 
-static avifBool avifParseToneMappedImageBox(avifGainMapMetadata * metadata, const uint8_t * raw, size_t rawLen, avifDiagnostics * diag)
+static avifBool avifParseToneMappedImageBox(avifGainMapMetadata * metadata,
```

Comment:
Optional/future: An alternative to adding the output parameter `isUnsupportedVersion` is to change the return type to `avifResult` and use the error code `AVIF_RESULT_NOT_IMPLEMENTED` to indicate an unsupported version.

This requires changing all the `AVIF_CHECK` macros used in the function.

Reply:
Done. I thought about that too but thought that it would make all the checks annoying, but actually extracting a sub function solves this issue. I think it's indeed cleaner now.